platform/mk2/hal/gps_skytraq/gps_device_skytraq.c

```diff
@@ -216,102 +228,121 @@ static uint8_t calculateChecksum(GpsMessage * msg)
+
                         msg->payload[i] = c;
-                    }
                 }
+                if (0 != len) {
+                        pr_trace("GPS: Failed to read payload\r\n");
+                        continue;
+                }
+
                 uint8_t checksum = 0;
-                if (!serial_read_byte
-                    (serial, &checksum, timeoutLen)) {
-                    result = GPS_MSG_TIMEOUT;
-                    break;
+                if (!serial_read_byte(serial, &checksum, timeoutLen)) {
+                        pr_trace("GPS: Failed to read checksum\r\n");
+                        continue;
                 }
-                uint8_t calculatedChecksum =
-                    calculateChecksum(msg);
-                if (calculatedChecksum == checksum) {
-                    uint8_t eos1 = 0, eos2 = 0;
-                    if (!
-                        (serial_read_byte
-                         (serial, &eos1, timeoutLen)
-                         && serial_read_byte(serial, &eos2,
-                                             timeoutLen))) {
-                        result = GPS_MSG_TIMEOUT;
-                        break;
-                    }
-                    if (eos1 == 0x0D && eos2 == 0x0A
-                        && msg->messageId ==
-                        expectedMessageId) {
-                        result = GPS_MSG_SUCCESS;
-                    } else {
-                        pr_trace_int_msg("Unexpected Id: ", msg->messageId);
-                        pr_trace_int_msg("Expected Id: ", expectedMessageId);
-                    }
+
+                uint8_t calculatedChecksum = calculateChecksum(msg);
+                if (calculatedChecksum != checksum) {
+                        pr_trace("GPS: Msg checksum mismatch\r\n");
+                        continue;
                 }
-            }
-        }
-        if (isTimeoutMs(timeoutStart, GPS_MSG_RX_WAIT_MS)) {
-            result = GPS_MSG_TIMEOUT;
+
+                uint8_t eos[2];
+                const bool read_eos =
+                        serial_read_byte(serial, eos + 0, timeoutLen) &&
+                        serial_read_byte(serial, eos + 1, timeoutLen);
+                if (!read_eos) {
+                        pr_trace("GPS: Failed to read EOS\r\n");
+                        continue;
+                }
+
+                if (eos[0] != 0x0D || eos[1] != 0x0A) {
+                        pr_trace("GPS: Invalid EOS\r\n");
+                        continue;
+                }
+
+                if (msg->messageId != expectedMessageId) {
+                        pr_trace_int_msg("GPS: Unexpected Message ID: ",
+                                        msg->messageId);
+                        continue;
+                }
+
+                return GPS_MSG_SUCCESS;
         }
-    }
-    return result;
 }
 
 static void sendSetFactoryDefaults(GpsMessage * gpsMsg, struct Serial * serial)
 {
+        poison_gps_message(gpsMsg);
```

Comment:
I suppose we don't.  Doesn't hurt to leave it in because if we see something fishy with the GPS we will know to go look for it.  Thoughts?